### PR TITLE
First stab at implementing decimal mode for ADC

### PR DESCRIPTION
**Pull request by omarandlorraine**
I've had a go at implementing the decimal mode because I saw that was missing. Actually, this is not always the right thing to do because of that Ricoh variant with no decimal mode. I would suggest some kind of conditional compilation here, but I don't know what you want to do about that. 

Actually it's my first time of writing any Rust, so I'd really appreciate if anyone has any critique. 

Some things I have not included:
- SBC. Yeah, I've only implemented decimal mode for addition but not for subtraction.
- No testing. I have not even tried this code (not figured out cargo and crates and whatnot), let alone thought about that test harness.

### CODE REVIEW

**Review by mre — APPROVED** [review]

**Comment by mre** [comment]
Looks good to me. Unit tests are passing. (I ran them manually because Github actions wasn't configured to run on pull requests, which I just fixed but I couldn't push to your branch to update it.)
Are you planning to add SBC as well before the merge?

**Comment by omarandlorraine** [comment]
What a coincidence, I was just now sitting down to do SBC.

Why can't you push to my branch? Can I give you permission somehow?

**Comment by mre** [comment]
> What a coincidence, I was just now sitting down to do SBC.

Cool!

> Why can't you push to my branch? Can I give you permission somehow?

[Sure!](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)

**Comment by omarandlorraine** [comment]
Ah, somehow that box is checked already, so from what I can see you should be able to push to my branch.

**Comment by omarandlorraine** [comment]
Okay I have implemented decimal SBC.

**Comment by mre** [comment]
Nice! Can you merge in master to trigger CI?

File: src/cpu.rs [hunk 1]
```diff
@@ -515,102 +515,116 @@ impl CPU {
...
+
+		debug!("accumulator: {}", self.registers.accumulator);
+	}
 
     fn and(&mut self, value: i8) {
         let a_after = self.registers.accumulator & value;
         self.load_accumulator(a_after);
     }
 
     // TODO: Implement binary-coded decimal
```
Comment (mre) [inline src/cpu.rs:569]: Guess that comment can go away?

File: src/cpu.rs [hunk 2]
```diff
@@ -766,6 +780,46 @@ mod tests {
...
+        assert_eq!(cpu.registers.status.contains(Status::PS_OVERFLOW), false);
+
+		cpu.add_with_carry(0x48);
+        assert_eq!(cpu.registers.accumulator, 0x00);
+        assert_eq!(cpu.registers.status.contains(Status::PS_CARRY), true);
+        assert_eq!(cpu.registers.status.contains(Status::PS_ZERO), true);
+        assert_eq!(cpu.registers.status.contains(Status::PS_NEGATIVE), false);
+        assert_eq!(cpu.registers.status.contains(Status::PS_OVERFLOW), true);
+
+		cpu.subtract_with_carry(0x48);
```
Comment (mre) [inline src/cpu.rs:809]: Do you think we should put the subtract tests into a separate test function? This way we could localize breakage of single operations more easily in the future.

File: src/cpu.rs (hunk 1 above)
Comment (omarandlorraine) [reply src/cpu.rs]: Yes it can

File: src/cpu.rs (hunk 2 above)
Comment (omarandlorraine) [reply src/cpu.rs:815]: It's true, I've separated it out

**Comment by omarandlorraine** [comment]
> Nice! Can you merge in master to trigger CI?

What do you mean here?

**Comment by mre** [comment]
> What do you mean here?

Sorry, I meant to do a `git pull origin/master && git push` just to get the latest changes from the master branch. Namely, I added continuous integration for pull requests now to run the tests. Those tests have been executed on the branch now and all looks good, so I'll go ahead and merge this. 
Thanks for your contribution @omarandlorraine. Would love to have you on the team if you're interested.

**Comment by omarandlorraine** [comment]
I couldn't do `git pull origin/master && git push` because `fatal: 'origin/master' does not appear to be a git repository`. I would guess I need to rebase my work on the new master that has CI. I might have made a mistake doing this work on the master branch of `omarandlorraine/mos6502`, but I guess I can work around that.

I'd be delighted to be part of the team, if the offer still stands!

**Pull request merged**